src/main/java/com/minecolonies/coremod/colony/jobs/JobNetherWorker.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Override
+    public CompoundNBT serializeNBT()
+    {
+        final CompoundNBT compound = super.serializeNBT();
+
+        @NotNull final ListNBT craftedList = new ListNBT();
+        craftedResults.forEach(item -> {
+            @NotNull final CompoundNBT itemCompound = item.serializeNBT();
+            craftedList.add(itemCompound);
+        });
+        compound.put(TAG_CRAFTED, craftedList);
+
+        @NotNull final ListNBT processedList = new ListNBT();
+        processedResults.forEach(item -> {
+            @NotNull final CompoundNBT itemCompound = item.serializeNBT();
+            processedList.add(itemCompound);
+        });
+        compound.put(TAG_PROCESSED, processedList);
+
+        compound.putBoolean(TAG_IN_NETHER, citizenInNether);
+        return compound;
+    }
+
+    @Override
+    public void deserializeNBT(final CompoundNBT compound)
+    {
+        super.deserializeNBT(compound);
+
+        final ListNBT craftedList = compound.getList(TAG_CRAFTED, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < craftedList.size(); ++i)
+        {
+            final CompoundNBT itemCompound = craftedList.getCompound(i);
+            craftedResults.add(ItemStack.of(itemCompound));
+        }
+
+        final ListNBT processedList = compound.getList(TAG_PROCESSED, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < processedList.size(); ++i)
+        {
+            final CompoundNBT itemCompound = processedList.getCompound(i);
+            processedResults.add(ItemStack.of(itemCompound));
+        }
+
+
+        if (compound.contains(TAG_IN_NETHER))
+        {
+            citizenInNether = compound.getBoolean(TAG_IN_NETHER);
+        }
+    }
+
+    @Override
+    public EntityAIWorkNether generateAI()
+    {
+        return new EntityAIWorkNether(this);
+    }
+
+    @NotNull
+    @Override
+    public BipedModelType getModel()
+    {
+        return BipedModelType.NETHER_WORKER;
+    }
+
+    @Override
+    public int getDiseaseModifier()
+    {
+        if(this.getCitizen().getEntity().isPresent() && this.getCitizen().getEntity().get().isInvisible())
+        {
+            return 0;
+        }
+        return super.getDiseaseModifier();
+    }
+
+    @Override
+    public int getIdleSeverity(boolean isDemand)
+    {
+        if(isDemand)
+        {
+            return super.getIdleSeverity(isDemand);
+        }
+        else
+        {
+            // Shorten the time for asking for materials. 
+            return 4;
+        }
+    }
+
+    /**
+     * Mark the worker as in the nether or not.
+     * @param away true if in the nether
+     */
+    public void setInNether(boolean away)
+    {
+        citizenInNether = away;
+    }
+
+    /**
+     * Check if the citizen is in the nether currently
+     */
+    public boolean isInNether()
+    {
+        return citizenInNether;
+    }
+
+    /**
+     * Get the queue of CraftedResults
+     */
+    public Queue<ItemStack> getCraftedResults()
```

Comment:
A bit unclean this way. Instead of getting and inserting into the list. Rather have a method here that adds to the respective lists.

Reply:
No, general poll and isEmpty are fine with getList (i.e. assume that the list is a copy). Just clear and add ItemStack ... stack is needed then.

Reply:
poll on a queue is both a read and a remove. It can't operate on a copy.

Reply:
Ah thought you meant with poll just a get and not pop.

Reply:
Hmm, then add a comment to the return method to make clear that the content of the queue is manipulated on return. And only the add a list of stacks method for add.

Reply:
done